cdi/src/test/java/io/smallrye/config/inject/ConfigInjectionTest.java

```diff
@@ -80,6 +83,9 @@ void converters() {
 
     @ApplicationScoped
     static class ConfigBean {
+        @Inject
+        @ConfigProperty(name = "reasons")
+        Map<Integer, String> reasons;
```

Comment:
I think we must add some more tests. Add more Map combinations, use a custom converter, etc.

Reply:
@radcortez I added some tests, let me know if it is not enough. Please note that it brings support of Map of List or Map of Set but only with `@ConfigProperty` but not in case of `@ConfigProperties`, if you expect the same support with `@ConfigProperties`, I think that it should be done in another ticket.

Reply:
@radcortez done